code/modules/economy/account.dm

```diff
@@ -30,6 +32,14 @@
 	account_balance += amt
 	if(account_balance < 0)
 		account_balance = 0
+	else if(account_balance > 1000000 && !is_bourgeois) // if we are now a millionaire, give the achievement
+		//So we currently only know what is *supposed* to be the real_name of the client's mob. If we can find them, we can get them this achievement. 
+		for(var/x in GLOB.player_list)
+			var/mob/M = x
+			if(M.real_name == account_holder)
+				SSachievements.unlock_achievement(/datum/achievement/cargo/bourgeois, M.client)
```

Comment:
Please add a check within the unlock_achievement proc to check if the client exists, its too easy to forget a ``if(whatever.client)``